### PR TITLE
Improve client test coverage #52

### PR DESCRIPTION
**Pull request by lzxm160**
work on #52

### CODE REVIEW

File: rpcmethod/client_test.go [hunk 1]
```diff
@@ -96,14 +99,11 @@ func TestServer_SendAction(t *testing.T) {
...
+	svr, err := NewRPCWithTLSEnabled(mainnet)
 	require.NoError(err)
-	actionHash := os.Getenv("actionHash")
-	actionActionInfoLen := os.Getenv("actionActionInfoLen")
-	actionActionNonce := os.Getenv("actionActionNonce")
-	if actionHash == "" || actionActionInfoLen == "" || actionActionNonce == "" {
-		t.Skip("skipping test; some params not set")
-	}
+	actionHash := "93de5923763c4ea79a01be023b49000838b1a4c22bdceed99dc23eeea8c9c757"
+	actionActionInfoLen := "1"
```
Comment (dustinxie) [inline rpcmethod/client_test.go]: directly use Equal(1, len()), Equal(27, Nonce()) below

can delete these 4 actionActionInfoLen, actionActionNonce, actionActionInfoLenInt, actionActionNonceInt

Comment (dustinxie) [reply rpcmethod/client_test.go]: also assert amount = 5 IOTX

File: rpcmethod/client_test.go [hunk 2]
```diff
@@ -126,17 +126,13 @@ func TestServer_GetAction(t *testing.T) {
...
-	}
+	getActionsByAddressActionHash := "633cf62ab47611476423d7416bb74395be9c9b602062074ac36744ddd31fd122"
 	request := &iotexapi.GetActionsRequest{
 		Lookup: &iotexapi.GetActionsRequest_ByAddr{
 			ByAddr: &iotexapi.GetActionsByAddressRequest{
-				Address: accountAddress,
+				Address: mainnetAddress,
 				Start:   1,
 				Count:   1,
 			},
```
Comment (dustinxie) [inline rpcmethod/client_test.go]: assert type = Execution, nonce, amount, Contract

File: rpcmethod/client_test.go [hunk 3]
```diff
@@ -173,30 +165,26 @@ func TestServer_GetUnconfirmedActionsByAddress(t *testing.T) {
...
+				BlkHash: mainnetBlockHash,
 				Start:   1,
 				Count:   10,
 			},
 		},
 	}
 	res, err := svr.GetActions(request)
 	require.NoError(err)
-	require.Equal(0, len(res.ActionInfo))
+	require.Equal(1, len(res.ActionInfo))
```
Comment (dustinxie) [inline rpcmethod/client_test.go]: this block contains 2 actions, how come 1 can pass here?

also assert details of these 2 actions:
https://www.iotexscan.io/block/89bbf8b1d3cbfb6020a1074a11c5430ef77eb220c00143dbd6f76d1cab94a1c2/action

File: rpcmethod/client_test.go [hunk 4]
```diff
@@ -222,102 +210,78 @@ func TestServer_GetBlockMetas(t *testing.T) {
...
 		},
 	}
 	res, err := svr.GetBlockMetas(request)
 	require.NoError(err)
 	require.Equal(1, len(res.BlkMetas))
 	blkPb := res.BlkMetas[0]
-	require.Equal(blk60801HashNumActionsInt, blkPb.NumActions)
-	require.Equal(blk60801HashTransferAmount, blkPb.TransferAmount)
+	require.Equal(int64(2), blkPb.NumActions)
+	require.Equal("0", blkPb.TransferAmount)
```
Comment (dustinxie) [inline rpcmethod/client_test.go]: assert res.Total = 1
assert other fields of blkPb

File: rpcmethod/client_test.go [hunk 5]
```diff
@@ -222,102 +210,78 @@ func TestServer_GetBlockMetas(t *testing.T) {
...
+	svr, err := NewRPCWithTLSEnabled(mainnet)
 	require.NoError(err)
 	res, err := svr.GetServerMeta(&iotexapi.GetServerMetaRequest{})
 	require.NoError(err)
-	getServerMetaPackageCommitID := os.Getenv("getServerMetaPackageCommitID")
-	if getServerMetaPackageCommitID == "" {
-		t.Skip("skipping test; some params not set")
-	}
-	require.Equal(getServerMetaPackageCommitID, res.GetServerMeta().PackageCommitID)
+	require.Equal("0810e5166d12c7ae06110cf6429f332c59585056", res.GetServerMeta().PackageCommitID)
```
Comment (dustinxie) [inline rpcmethod/client_test.go]: commit ID can change, use require.NotEqual(""), also assert for other string fields

for BuildTime, can assert BuildTime >= the time get here

File: rpcmethod/client_test.go [hunk 6]
```diff
@@ -222,102 +210,78 @@ func TestServer_GetBlockMetas(t *testing.T) {
...
 	})
 	require.NoError(err)
 	require.NotNil(out)
 	val, ok := big.NewInt(0).SetString(string(out.Data), 10)
 	require.True(ok)
-	expected, ok := new(big.Int).SetString(accountAddressUnclaimedBalance, 10)
+	expected, ok := new(big.Int).SetString("39860707937452088904761", 10)
 	require.True(ok)
-	require.Equal(0, val.Cmp(expected))
+	require.Equal(1, val.Cmp(expected))
```
Comment (dustinxie) [inline rpcmethod/client_test.go]: cannot use Equal, unclaimed balance can change
i think change to read these 2 states:

1. active BP list, assert number > 0
2. gravity chain height, assert > 7769900

File: rpcmethod/client_test.go [hunk 7]
```diff
@@ -222,102 +210,78 @@ func TestServer_GetBlockMetas(t *testing.T) {
...
-	request := &iotexapi.GetReceiptByActionRequest{ActionHash: actionHash}
+	request := &iotexapi.GetReceiptByActionRequest{ActionHash: mainnetReceiptHash}
 	res, err := svr.GetReceiptByAction(request)
 	require.NoError(err)
 	require.NotNil(res)
 	receiptPb := res.ReceiptInfo.Receipt
-	require.Equal(uint64(3151), receiptPb.Status)
-	require.Equal(getReceiptByActionBlkHeightInt, receiptPb.BlkHeight)
+	require.Equal(uint64(1), receiptPb.Status)
+	require.Equal(uint64(56664), receiptPb.BlkHeight)
```
Comment (dustinxie) [inline rpcmethod/client_test.go]: assert GasConsumed, ContractAddress, len(Logs)=1

File: rpcmethod/client_test.go (hunk 1 above)
Comment (lzxm160) [reply rpcmethod/client_test.go]: sure

**Pull request merged**